portability_builtins.hpp

```diff
@@ -27,7 +27,25 @@
 namespace unodb::detail {
 
 /// Reverse the order of bytes in \a x.
-[[nodiscard, gnu::pure]] UNODB_DETAIL_CONSTEXPR_NOT_MSVC std::uint64_t bswap(
+[[nodiscard, gnu::pure]] UNODB_DETAIL_CONSTEXPR_NOT_MSVC std::uint16_t bswap16(
```

Comment:
Don't introduce 16/32/64 suffixes - templatize and if constexpr on the width inside. This file has an example